bitnami/apisix/templates/tls-secret.yaml

```
@@ -0,0 +1,142 @@
+{{- $ca := genCA "apisix-ca" 365 }}
+{{- if and .Values.controlPlane.enabled (not .Values.controlPlane.tls.existingSecret) }}
+{{/* For the control plane, it is mandatory to have TLS for the config server */}}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ printf "%s-tls" (include "apisix.control-plane.fullname" .) }}
+  namespace: {{ include "common.names.namespace" . | quote }}
+  labels: {{- include "common.labels.standard" . | nindent 4 }}
+    app.kubernetes.io/part-of: apisix
+    app.kubernetes.io/component: control-plane
+    {{- if .Values.commonLabels }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
+    {{- end }}
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+type: kubernetes.io/tls
+data:
+  {{- if .Values.controlPlane.tls.autoGenerated }}
+  {{- $cert := genSignedCert (include "apisix.control-plane.fullname" .) nil (list (include "apisix.control-plane.fullname" .) (printf "%s.%s" (include "apisix.control-plane.fullname" .) (include "common.names.namespace" .)) (printf "%s.%s.svc" (include "apisix.control-plane.fullname" .) (include "common.names.namespace" .)) (printf "%s.%s.svc.%s" (include "apisix.control-plane.fullname" .) (include "common.names.namespace" .) .Values.clusterDomain)) 365 $ca }}
+  {{ .Values.controlPlane.tls.certFilename }}: {{ include "common.secrets.lookup" (dict "secret" (printf "%s-tls" (include "apisix.control-plane.fullname" .)) "key" .Values.controlPlane.tls.certFilename "defaultValue" $cert.Cert "context" $) }}
+  {{ .Values.controlPlane.tls.certKeyFilename }}: {{ include "common.secrets.lookup" (dict "secret" (printf "%s-tls" (include "apisix.control-plane.fullname" .)) "key" .Values.controlPlane.tls.certKeyFilename "defaultValue" $cert.Key "context" $) }}
+  {{ .Values.controlPlane.tls.certCAFilename }}: {{ include "common.secrets.lookup" (dict "secret" (printf "%s-tls" (include "apisix.control-plane.fullname" .)) "key" .Values.controlPlane.tls.certCAKeyFilename "defaultValue" $ca.Cert "context" $) }}
```

The lookup key is wrong:

```suggestion
  {{ .Values.controlPlane.tls.certCAFilename }}: {{ include "common.secrets.lookup" (dict "secret" (printf "%s-tls" (include "apisix.control-plane.fullname" .)) "key" .Values.controlPlane.tls.certCAFilename "defaultValue" $ca.Cert "context" $) }}
```